recipes-core/initrdscripts/initramfs-stubdomain/init.sh

```diff
@@ -77,8 +77,18 @@ export INTEL_DBUS=1
 echo "Starting qemu directly."
 target="$( xenstore-read target )"
 vm_uuid="$( xenstore-read /local/domain/${target}/vm )"
-dmargs="$( xenstore-read ${vm_uuid}/image/dmargs )"
+
+dmargs=$( xenstore-read $(xenstore-list -p "$vm_uuid/image/dm-argv" | sort ))
 echo "target $target vm_uuid $vm_uuid"
 echo "Invoking qemu with dmargs       = ${dmargs}"
-/usr/bin/qemu-system-i386 ${dmargs}
+
+/usr/bin/qemu-system-i386 ${dmargs} -chardev socket,server,nowait,path=/tmp/qemu.qmp,id=m2 -mon chardev=m2,mode=control &
+
+device_model="device-model/$target"
+
+echo "Starting vchan-socket-proxy"
+vchan-socket-proxy -m server 0 $device_model/qmp-vchan /tmp/qemu.qmp &
+
+wait
```

Comment:
In the current implementation, QEMU is a blocking process. On HVM domain shutdown, somehow it exits cleanly and then the stubdom ```poweroff -f```'s.  With the ```wait``` and the backgrounded QEMU process, will the stubdom still shutdown in the same fashion?

Reply:
The poweroff hasn't seemed necessary in my testing.  I think QEMU exists on QMP command.  Later commits add the console, which has the init 1 shell waiting.  Things seem to work fine and there aren't any messages about the stubdom crashing.

Reply:
Ok, sounds good.